docs/articles/documentation/recipes/basics/test-file-upload.md

```diff
@@ -0,0 +1,83 @@
+---
+layout: docs
+title: Test File Upload
+permalink: /documentation/recipes/basics/test-file-upload.html
+---
+# Test File Upload
+
+This recipe shows how to test your application's file upload functionality with TestCafe.
+
+[Full Example Code](https://github.com/DevExpress/testcafe-examples/tree/master/examples/upload-files)
+
+Assume the following HTML:
+
+```html
+<body>
+    <form action="http://localhost:3000/upload" method="post" enctype="multipart/form-data">
+        <input id="upload-input" type="file" name="files" multiple />
+        <input type="submit" id="upload-btn" value="Upload Files"/>
+    </form>
+</body>
+```
+
+TestCafe's file upload emulation is differrent from what a real user would do to upload files.
+
+To manually upload files on this page, the user needs to perform the following steps:
```

Comment:
```suggestion
To manually upload files to this page, the user needs to perform the following steps:
```

Reply:
How about:
To upload files, users should perform the following steps:

Reply:
In this article, we focus on key differences between two ways to upload files: the "manual way" and the "TestCafe way". It's necessary to draw users' attention to the following fact: 

_In TestCafe, we upload files differently than we would without TestCafe._

Can we rewrite the section like so:

> Without TestCafe, users should perform the following steps to upload files:
> 
> * Click the 'Choose File' button that opens the 'Choose file' dialog in the browser
> * Select the files
> * Begin the upload
> 
> Since TestCafe cannot interact with the native dialog, it uses a unique approach, which is to manipulate the `<input type=file>` elements' state with the `.setFilesToUpload` action.

we can use technique instead if it'd be better:
> Since TestCafe cannot interact with the native dialog, it uses a unique **technique**, which is to manipulate the `<input type=file>` elements' state with the `.setFilesToUpload` action.

Reply:
-"When manually uploading files (without the use of TestCafe), users should perform..."

-"Since TestCafe cannot interact with the native dialog, it manipulates the <input type=file> elements' state with the .setFilesToUpload action."

Manipulate doesn't work here. You can't manipulate something WITH something. Did you mean combines? Replaces? Joins?

Reply:
This action fills the element with files.

Does this work:

> Since TestCafe cannot interact with the native dialog, it uses the .setFilesToUpload action to manipulate the `<input type=file>` elements' state.

Reply:
yes, just change the endingg
to manipulate the element state

Reply:
> TestCafe cannot interact with the native dialog. It uses the .setFilesToUpload action to change the <input type=file> elements' state.

Reply:
Great!